java/vector/src/main/java/org/apache/arrow/vector/IntervalMonthDayNanoVector.java

```diff
@@ -0,0 +1,440 @@
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.TransferPair;
+
+/**
+ * IntervalMonthDayNanoVectorimplements a fixed width vector (16 bytes) of
+ * interval (month, days and nanoseconds) values which could be null.
+ * A validity buffer (bit vector) is maintained to track which elements in the
+ * vector are null.
+ */
+public final class IntervalMonthDayNanoVector extends BaseFixedWidthVector {
+  public static final byte TYPE_WIDTH = 16;
+  private static final byte DAY_OFFSET = 4;
+  private static final byte NANOSECOND_OFFSET = 8;
+  private final FieldReader reader;
+
+
+  /**
+   * Instantiate a IntervalMonthDayNanoVector. This doesn't allocate any memory for
+   * the data in vector.
+   *
+   * @param name name of the vector
+   * @param allocator allocator for memory management.
+   */
+  public IntervalMonthDayNanoVector(String name, BufferAllocator allocator) {
+    this(name, FieldType.nullable(MinorType.INTERVALDAY.getType()), allocator);
+  }
+
+  /**
+   * Instantiate a IntervalMonthDayNanoVector. This doesn't allocate any memory for
+   * the data in vector.
+   *
+   * @param name name of the vector
+   * @param fieldType type of Field materialized by this vector
+   * @param allocator allocator for memory management.
+   */
+  public IntervalMonthDayNanoVector(String name, FieldType fieldType, BufferAllocator allocator) {
+    this(new Field(name, fieldType, null), allocator);
+  }
+
+  /**
+   * Instantiate a IntervalMonthDayNanoVector. This doesn't allocate any memory for
+   * the data in vector.
+   *
+   * @param field field materialized by this vector
+   * @param allocator allocator for memory management.
+   */
+  public IntervalMonthDayNanoVector(Field field, BufferAllocator allocator) {
+    super(field, allocator, TYPE_WIDTH);
+    reader = new IntervalMonthDayNanoReaderImpl(IntervalMonthDayNanoVector.this);
+  }
+
+  /**
+   * Get a reader that supports reading values from this vector.
+   *
+   * @return Field Reader for this vector
+   */
+  @Override
+  public FieldReader getReader() {
+    return reader;
+  }
+
+  /**
+   * Get minor type for this vector. The vector holds values belonging
+   * to a particular type.
+   *
+   * @return {@link org.apache.arrow.vector.types.Types.MinorType}
+   */
+  @Override
+  public MinorType getMinorType() {
+    return MinorType.INTERVALMONTHDAYNANO;
+  }
+
+
+  /*----------------------------------------------------------------*
+   |                                                                |
+   |          vector value retrieval methods                        |
+   |                                                                |
+   *----------------------------------------------------------------*/
+
+  /**
+   * Given a data buffer, get the number of months stored at a particular position
+   * in the vector.
+   *
+   * <p>This method should not be used externally.
```

Comment:
If it is not intended to be used externally, then it should not be declared "public"?

Reply:
This unfortunately is a pattern established for integration testing which lives in a separate package.